packages/web3-eth/src/utils/transaction_builder.ts

```diff
@@ -238,7 +236,21 @@ export async function defaultTransactionBuilder<ReturnType = Transaction>(option
 				ETH_DATA_FORMAT,
 			)),
 		};
-
+	if (
+		(isNullish(populatedTransaction.gas) || isNullish(populatedTransaction.gasLimit)) &&
+		options.fillGas
+	) {
+		const fillGas = await estimateGas(
+			options.web3Context,
+			populatedTransaction,
+			'latest',
+			ETH_DATA_FORMAT,
+		);
+		populatedTransaction = {
+			...populatedTransaction,
+			gas: format({ format: 'uint' }, fillGas as Numbers, ETH_DATA_FORMAT),
```

Comment:
can you point me in 1.x where we are setting gasLimit ( gas ) ?

Reply:
Ah I don't believe we set it in 1.x sorry for the confusion, this is [available](https://github.com/ethers-io/ethers.js/blob/1dd542721d3a9e74d10a92847200f4b2aa5da618/src.ts/providers/abstract-signer.ts#L105) in ethers and would be a solution to solving the transaction evm revert errors [reported](https://github.com/web3/web3.js/issues/6249) by users.